storm-core/src/jvm/org/apache/storm/scheduler/EvenScheduler.java

```diff
@@ -0,0 +1,177 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.storm.scheduler;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+import java.util.TreeMap;
+
+import org.apache.storm.utils.Utils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.collect.Sets;
+
+public class EvenScheduler implements IScheduler {
+    private static final Logger LOG = LoggerFactory.getLogger(EvenScheduler.class);
+
+    public static List<WorkerSlot> sortSlots(List<WorkerSlot> availableSlots, Cluster cluster) {
+        if (availableSlots != null && availableSlots.size() > 0) {
+            // group by node
+            Map<String, List<WorkerSlot>> slotGroups = new TreeMap<String, List<WorkerSlot>>();
+            for (WorkerSlot slot : availableSlots) {
+                String host = cluster.getHost(slot.getNodeId());
+                List<WorkerSlot> slots = slotGroups.get(host);
+                if (slots == null) {
+                    slots = new ArrayList<WorkerSlot>();
+                    slotGroups.put(host, slots);
+                }
+                slots.add(slot);
+            }
+
+            // sort by port
+            for (List<WorkerSlot> slots : slotGroups.values()) {
+                Collections.sort(slots, new Comparator<WorkerSlot>() {
+                    @Override
+                    public int compare(WorkerSlot o1, WorkerSlot o2) {
+                        return o1.getPort() - o2.getPort();
+                    }
+                });
+            }
+
+            // sort by count
+            List<List<WorkerSlot>> list = new ArrayList<List<WorkerSlot>>(slotGroups.values());
+            Collections.sort(list, new Comparator<List<WorkerSlot>>() {
+                @Override
+                public int compare(List<WorkerSlot> o1, List<WorkerSlot> o2) {
+                    return o2.size() - o1.size();
+                }
+            });
+
+            return Utils.interleaveAll(list);
+        }
+
+        return null;
+    }
+
+    public static Map<WorkerSlot, List<ExecutorDetails>> getAliveAssignedWorkerSlotExecutors(Cluster cluster, String topologyId) {
+        SchedulerAssignment existingAssignment = cluster.getAssignmentById(topologyId);
+        Map<ExecutorDetails, WorkerSlot> executorToSlot = null;
+        if (existingAssignment != null) {
+            executorToSlot = existingAssignment.getExecutorToSlot();
+        }
+
+        Map<WorkerSlot, List<ExecutorDetails>> result = new HashMap<WorkerSlot, List<ExecutorDetails>>();
+        if (executorToSlot != null) {
+            for (Entry<ExecutorDetails, WorkerSlot> entry : executorToSlot.entrySet()) {
```

Comment:
You can use Utils.reverseMap
https://github.com/apache/storm/blob/1a445164a96afdc21eac29856f2da569f331e855/storm-core/src/jvm/org/apache/storm/utils/Utils.java#L1715